nwbwidgets/ophys.py

```diff
@@ -113,52 +112,120 @@ def show_plane_segmentation_2d(plane_seg: PlaneSegmentation, color_wheel=color_w
+                    legendgroup=str(plane_seg[color_by][i]),
+                    showlegend=show_leg,
+                    text=hovertext,
+                    hovertext='text',
+                    line=dict(width=.5),
+                    **kwargs
+                )
+            )
+            fig.update_layout(
+                width=700, height=500,
+                margin=go.layout.Margin(l=60, r=60, b=60, t=60, pad=1),
+                plot_bgcolor="rgb(245, 245, 245)",
             )
-        )
-        fig.update_layout(
-            width=700, height=500,
-            margin=go.layout.Margin(l=60, r=60, b=60, t=60, pad=1),
-            plot_bgcolor="rgb(245, 245, 245)",
-        )
+    else:
+        fig.update_traces(visible=False)
+        for i in range(len(fig.data)):
+            fig.data[i].line.color = color_wheel[np.where(cats == plane_seg[color_by][i])[0][0]]
+        fig.update_traces(visible=True)
+        """kwargs.update_traces()
+Here change the traces?
+"""
+        # fig.data = None
+
     return fig
 
 
+"""NEW CLASS"""
+
+
+class plane_segmentation_2d_widget(widgets.VBox):
+    def __init__(self, plane_seg: PlaneSegmentation, color_wheel=color_wheel, color_by='neuron_type', threshold=.01,
+                 fig=None, **kwargs):
+        super().__init__()  # is the underlying structure an HBox?
+        self.categorical_columns = infer_categorical_columns(plane_seg)
+        self.plane_seg = plane_seg
+
+        if len(self.categorical_columns) == 1:
+            self.color_by = list(self.categorical_columns.keys())[0]  # changing local variables to instance variables?
+            self.children = [show_plane_segmentation_2d(plane_seg, color_by=color_by, **kwargs)]
+        elif len(self.categorical_columns) > 1:
+            self.cat_controller = widgets.Dropdown(options=list(self.categorical_columns), description='color by')
+            self.out_fig = show_plane_segmentation_2d(plane_seg, color_by=self.cat_controller.value, **kwargs)
+
+            def on_change(change):  # get rid of outfig as a parameter
+                if change['new'] and isinstance(change['new'], dict):
+                    ind = change['new']['index']
+                    if isinstance(ind, int):
+                        color_by = change['owner'].options[ind]
+                        self.update_trace_plane_segmentation_2d(color_by)
+                        """call an update function here, no longer want to rewrite"""
+                    # show_plane_segmentation_2d(self.plane_seg, color_by=color_by, fig=self.out_fig, **kwargs) #replace outfig with instance variable
+
+            self.cat_controller.observe(on_change)
+            self.children = [self.cat_controller, self.out_fig]
+        else:
+            self.children = [show_plane_segmentation_2d(self.plane_seg, color_by=None, **kwargs)]
+
+    def update_trace_plane_segmentation_2d(self, color_by):
+        display = self.children[1]
```

Comment:
See new Peekaboo class [here](https://github.com/NeurodataWithoutBorders/nwb-jupyter-widgets/blob/6f4514ec168d7cf8088e7f9ba4a8c652e3f2b2f4/nwbwidgets/utils/plotly.py#L78-L95). If it's not clear how to use it, see [this article](https://jeffknupp.com/blog/2016/03/07/python-with-context-managers/)